tests/test_type_resolution.py

```diff
@@ -149,3 +149,21 @@ def func2(bar: "Bar", foo: "Foo"):  # noqa
 
     assert sig2.parameters["foo"].annotation == Foo
     assert sig2.parameters["bar"].annotation == "Bar"
+
+
+GlobalThing = int
+
+
+def test_type_resolved_signature_mixed_global() -> None:
+    """Test that we can resolve a mix of global annotations and missing forward refs."""
+
+    def myfun(a: "unknown", b: "GlobalThing"):  # type: ignore  # noqa
```

Comment:
I think to test it properly we need to include `from __future__ import annotations`

Reply:
The effect of that is simply to convert all type annotations to strings during parsing.  So using `"GlobalThing"` (in quotes) has the same effect.

These tests failed on main

Reply:
https://peps.python.org/pep-0563/#implementation

Reply:
Of course, thanks!